File: spring22/T2/NLP/ruche_examples-master/examples_formation/hello_omp.f90
program main

  use omp_lib

  implicit none

  integer:: id

  write ( *, '(a,i8)' ) &
    'The number of processors available = ', omp_get_num_procs()
  write ( *, '(a,i8)' ) &
    'The number of threads available    = ', omp_get_max_threads()

!$omp parallel private ( id )
  id = omp_get_thread_num()
  write ( *, '(a,i8,a,i8)' ) 'HELLO from process ', id
!$omp end parallel

end program main

File: spring22/T2/NLP/ruche_examples-master/examples_formation/hello_omp.c
# include <stdlib.h>
# include <stdio.h>
# include <omp.h>

int main ( int argc, char *argv[] );

/******************************************************************************/

int main ( int argc, char *argv[] )
{
  int id;

  printf ("Number of processors available = %d\n",omp_get_num_procs());
  printf ("Number of threads =              %d\n",omp_get_max_threads());

#pragma omp parallel private(id)
  {
    id = omp_get_thread_num();
    printf ("Hello from process %d\n",id);
  }
  return 0;
}
